src/components/attendee/VideoComms/internal/TwilioImplementation.tsx

```diff
@@ -0,0 +1,378 @@
+  name: string;
+  attach: () => void;
+  detach: () => void;
+  isEnabled: boolean;
+}
+
+const wrapTrack = <Kind, TrackType extends Tracklike>(
+  track: TrackType,
+  kind: Kind,
+  id: TrackId
+) => ({
+  kind,
+  id,
+  attach: track.attach.bind(track),
+  detach: track.detach.bind(track),
+  twilioTrack: track,
+  enabled: track.isEnabled,
+  sourceType: getTrackSource(track),
+});
+
+const wrapRemoteVideoTrack = (track: Twilio.RemoteVideoTrack): VideoTrack =>
+  wrapTrack(track, "video", track.sid as TrackId);
+
+const wrapLocalVideoPublication = (
+  publication: Twilio.LocalVideoTrackPublication
+): VideoTrack =>
+  wrapTrack(publication.track, "video", publication.trackSid as TrackId);
+
+const wrapRemoteAudioTrack = (track: Twilio.RemoteAudioTrack): AudioTrack =>
+  wrapTrack(track, "audio", track.sid as TrackId);
+
+const wrapLocalAudioPublication = (
+  publication: Twilio.LocalAudioTrackPublication
+): AudioTrack =>
+  wrapTrack(publication.track, "audio", publication.trackSid as TrackId);
+
+const remotePublicationsToTracks = (
+  publications: Map<String, Twilio.RemoteVideoTrackPublication>
+): VideoTrack[] => {
+  return Array.from(publications.values())
+    .map((vt) => vt.track && wrapRemoteVideoTrack(vt.track))
+    .filter(isDefined);
+};
+
+const remotePublicationsToAudioTracks = (
+  publications: Map<String, Twilio.RemoteAudioTrackPublication>
+): AudioTrack[] => {
+  return Array.from(publications.values())
+    .map((at) => at.track && wrapRemoteAudioTrack(at.track))
+    .filter(isDefined);
+};
+
+const localPublicationsToVideoTracks = (
+  publications: Map<String, Twilio.LocalVideoTrackPublication>
+): VideoTrack[] => {
+  return Array.from(publications.values())
+    .map((p) => p.track && wrapLocalVideoPublication(p))
+    .filter(isDefined);
+};
+
+const localPublicationsToAudioTracks = (
+  publications: Map<String, Twilio.LocalAudioTrackPublication>
+): AudioTrack[] => {
+  return Array.from(publications.values())
+    .map((p) => p.track && wrapLocalAudioPublication(p))
+    .filter(isDefined);
+};
+
+const wrapLocalParticipant = (room: Twilio.Room) => ({
+  id: room.localParticipant.sid,
+  sparkleId: room.localParticipant.identity,
+  audioTracks: localPublicationsToAudioTracks(
+    room.localParticipant.audioTracks
+  ),
+  videoTracks: localPublicationsToVideoTracks(
+    room.localParticipant.videoTracks
+  ),
+});
+
+/**
+ * This provides an interface over the top of Twilio. It is designed to
+ * work outside of React as much as possible to avoid the complexity of dealing
+ * with hooks.
+ *
+ * Instead, the method triggerStatusUpdate is used to let React know that the
+ * state of the video call has changed.
+ *
+ * It is expected that this is not used directly by developers and all
+ * interactions happen via the hooks and components provided in VideoComms.
+ */
+export const TwilioImplementation = (
+  onStateUpdateCallback: StateUpdateCallback
+) => {
+  let room: Twilio.Room | undefined;
```

Comment:
maybe it's worthwhile to give this an explicit value of `undefined`, so it looks like `let room: Twilio.Room | undefined = undefined`, kinda made me wonder for a minute if everything is alright with this line written like this